packages/vite/src/client/client.ts

```diff
@@ -111,7 +111,7 @@ function setupWebSocket(
 }
 
 function cleanUrl(pathname: string): string {
-  const url = new URL(pathname, location.toString())
+  const url = new URL(pathname, location.protocol === 'blob:' ? location.pathname : location.toString());
```

Comment:
```suggestion
  const url = new URL(pathname, 'http://vitejs.dev');
```
https://github.com/vitejs/vite/blob/74dde790ffcd982fa9824ed38b8ba5afccb8f609/packages/vite/src/client/client.ts#L447
Because we don't use the base part, let's do the same with `injectQuery`.

Reply:
I agree your suggest to change the code same with the `injectQuery`. However, this is out of topic, i suggest  to change the regex at previous line into `/[\?#].*$/` because without slash `\`, it will match the string `'#?'` not match query or fragment of URL. https://github.com/vitejs/vite/blob/74dde790ffcd982fa9824ed38b8ba5afccb8f609/packages/vite/src/client/client.ts#L446